Index: ofed_kernel-2.6.33/drivers/infiniband/ulp/sdp/sdp.h
===================================================================
--- ofed_kernel-2.6.33.orig/drivers/infiniband/ulp/sdp/sdp.h	2011-01-05 13:27:06.000000000 +0200
+++ ofed_kernel-2.6.33/drivers/infiniband/ulp/sdp/sdp.h	2011-01-05 13:36:19.000000000 +0200
@@ -42,12 +42,12 @@
 #endif
 #endif
 
-#define inet_num(sk) inet_sk(sk)->num
-#define inet_sport(sk) inet_sk(sk)->sport
-#define inet_dport(sk) inet_sk(sk)->dport
-#define inet_saddr(sk) inet_sk(sk)->saddr
-#define inet_daddr(sk) inet_sk(sk)->daddr
-#define inet_rcv_saddr(sk) inet_sk(sk)->rcv_saddr
+#define inet_num(sk) inet_sk(sk)->inet_num
+#define inet_sport(sk) inet_sk(sk)->inet_sport
+#define inet_dport(sk) inet_sk(sk)->inet_dport
+#define inet_saddr(sk) inet_sk(sk)->inet_saddr
+#define inet_daddr(sk) inet_sk(sk)->inet_daddr
+#define inet_rcv_saddr(sk) inet_sk(sk)->inet_rcv_saddr
 
 #define sk_ssk(ssk) ((struct sock *)ssk)
 
