diff -rup old/drivers/infiniband/hw/cxgb3/iwch_cm.c new/drivers/infiniband/hw/cxgb3/iwch_cm.c
--- old/drivers/infiniband/hw/cxgb3/iwch_cm.c	2011-01-31 12:19:40.000000000 -0600
+++ new/drivers/infiniband/hw/cxgb3/iwch_cm.c	2011-01-31 12:20:05.000000000 -0600
@@ -1397,7 +1397,7 @@ static int pass_accept_req(struct t3cdev
 		       __func__);
 		goto reject;
 	}
-	dst = &rt->u.dst;
+	dst = &rt->dst;
 	l2t = t3_l2t_get(tdev, dst->neighbour, dst->neighbour->dev);
 	if (!l2t) {
 		printk(KERN_ERR MOD "%s - failed to allocate l2t entry!\n",
@@ -1965,7 +1965,7 @@ int iwch_connect(struct iw_cm_id *cm_id,
 		err = -EHOSTUNREACH;
 		goto fail3;
 	}
-	ep->dst = &rt->u.dst;
+	ep->dst = &rt->dst;
 
 	/* get a l2t entry */
 	ep->l2t = t3_l2t_get(ep->com.tdev, ep->dst->neighbour,
