diff --git a/drivers/net/mlx4/en_tx.c b/drivers/net/mlx4/en_tx.c
index 1f1ccb7..f8cffbb 100644
--- a/drivers/net/mlx4/en_tx.c
+++ b/drivers/net/mlx4/en_tx.c
@@ -657,7 +657,7 @@ u16 mlx4_en_select_queue(struct net_device *dev, struct sk_buff *skb)
 	}
 
 	entry->cnt++;
-	if (skb->len > MLX4_EN_SMALL_PKT_SIZE)
+	if (skb->len > MLX4_EN_SMALL_TX)
 		entry->big_pkts++;
 	else
 		entry->small_pkts++;
diff --git a/drivers/net/mlx4/mlx4_en.h b/drivers/net/mlx4/mlx4_en.h
index 79f6eea..95ff349 100644
--- a/drivers/net/mlx4/mlx4_en.h
+++ b/drivers/net/mlx4/mlx4_en.h
@@ -142,6 +142,7 @@ enum {
 #define MLX4_EN_MIN_TX_SIZE	(4096 / TXBB_SIZE)
 
 #define MLX4_EN_SMALL_PKT_SIZE		64
+#define MLX4_EN_SMALL_TX		128
 #define MLX4_EN_TX_HASH_SIZE		256
 #define MLX4_EN_TX_HASH_MASK		(MLX4_EN_TX_HASH_SIZE - 1)
 #define MLX4_EN_NUM_HASH_RINGS		4
